[iOS] Background for multi-file upload icon is drawn incorrectly

rr-codes opened the pull request:
#### b6c00b300dfe5f6f0c74db1f20fa20363caf7896
<pre>
[iOS] Background for multi-file upload icon is drawn incorrectly
<a href="https://bugs.webkit.org/show_bug.cgi?id=246063">https://bugs.webkit.org/show_bug.cgi?id=246063</a>
rdar://100784312

Reviewed by Tim Horton and Aditya Keerthi.

Currently, for file inputs with multiple selected files, iOS tries to draw
an icon that has a gradient colored background. However, the gradient isn&apos;t
actually drawn as a result of iOS using the GPU process.

This PR resolves that by replacing the gradient with a simple base color,
as the gradient no longer makes sense anyways since the button doesn’t have
one either.

* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::paintFileUploadIconDecorations):

Canonical link: <a href="https://commits.webkit.org/255202@main">https://commits.webkit.org/255202@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d5ddb6a53225d979a85953dce638f4058482e483

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91576 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/748 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22196 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101283 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/161350 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95581 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/899 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29431 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83898 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97629 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97234 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/528 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78257 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27408 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82452 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/82032 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70447 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35700 "Built successfully") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16062 "Found 1 new test failure: webanimations/accelerated-web-animation-with-single-interval-and-easing-y-axis-above-1.html (failure)") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33464 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17156 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3606 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37294 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39931 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39200 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36273 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebCore/rendering/RenderThemeIOS.mm:
```diff
@@ -1328,21 +1328,10 @@ static bool shouldUseConvexGradient(const Color& backgroundColor)
-            CGContextRef cgContext = paintInfo.context().platformContext();
-            paintInfo.context().clip(thumbnailRect);
-            if (shouldUseConvexGradient(backgroundImageColor))
-                drawAxialGradient(cgContext, gradientWithName(ConvexGradient), thumbnailRect.location(), FloatPoint(thumbnailRect.x(), thumbnailRect.maxY()), LinearInterpolation);
```
Are there any remaining clients of ConvexGradient, ShadeGradient, or ShineGradient, or can we follow this deletion to those?

rr-codes replied:
yes, for example all are used within `paintPushButtonDecorations`

pxlcoder replied:
That code can probably be removed now too... but definitely a separate PR, and unrelated to the work you've been doing.

rr-codes committed:
[iOS] Background for multi-file upload icon is drawn incorrectly
https://bugs.webkit.org/show_bug.cgi?id=246063
rdar://100784312

Reviewed by Tim Horton and Aditya Keerthi.

Currently, for file inputs with multiple selected files, iOS tries to draw
an icon that has a gradient colored background. However, the gradient isn't
actually drawn as a result of iOS using the GPU process.

This PR resolves that by replacing the gradient with a simple base color,
as the gradient no longer makes sense anyways since the button doesn’t have
one either.

* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::paintFileUploadIconDecorations):

Canonical link: https://commits.webkit.org/255202@main
